crates/sui-core/src/authority_active/checkpoint_executor/mod.rs

```diff
@@ -0,0 +1,252 @@
+                                    next_checkpoint_clone.clone(),
+                                    state.clone(),
+                                    store.clone(),
+                                ).await {
+                                    error!(
+                                        "Error while executing checkpoint, will retry in 1s: {:?}",
+                                        err
+                                    );
+                                    tokio::time::sleep(Duration::from_secs(1)).await;
+                                    metrics.checkpoint_exec_errors.inc();
+                                }
+
+                                (seq_num, next_checkpoint_clone)
+                            }));
+
+                            // Last checkpoint of epoch
+                            if let Some(next_committee) = next_checkpoint.next_epoch_committee() {
+                                // Unwrap ok because reconfig failure is unrecoverable
+                                self.reconfig(next_committee).await.unwrap();
+                            }
+
+                            if finished.len() >= task_limit {
+                                break;
+                            }
+                        }
+                    }
+                    // In this case, messages in the mailbox have been overwritten
+                    // as a result of lagging too far behind. We can simply continue the loop,
+                    // as the next call to recv() should return the overwritten checkpoint and we
+                    // will attempt to execute all checkpoints from (highest_executed_checkpoint + 1)
+                    // up to the result of the next recv() call.
+                    Err(RecvError::Lagged(num_skipped)) => {
+                        warn!(
+                            "Checkpoint Execution Recv channel overflowed {:?} messages",
+                            num_skipped,
+                        );
+                        self.metrics
+                            .checkpoint_exec_recv_channel_overflow
+                            .inc_by(num_skipped);
+                    }
+                    Err(RecvError::Closed) => {
+                        info!("Checkpoint Execution Sender (StateSync) closed channel");
+                        break;
+                    }
+                },
+            }
+        }
+    }
+
+    async fn reconfig(&self, _next_epoch_committee: &[(AuthorityName, StakeUnit)]) -> SuiResult {
+        // TODO(william) call into Reconfig
+        Ok(())
+    }
+
+    fn update_lag_metric(&self, highest_executed: CheckpointSequenceNumber) -> SuiResult {
+        let highest_synced = self
+            .checkpoint_store
+            .get_highest_synced_checkpoint_seq_number()?
+            .unwrap_or(0);
+
+        let diff = highest_synced - highest_executed;
+        self.metrics.checkpoint_exec_lag.set(diff as i64);
+        Ok(())
+    }
+}
```

Comment:
We may not explicitly need this metric since we'll have metrics for both highest_synced and highest_executed and can rely on creating this metrics in grafana